app/controllers/pages_controller.rb

```diff
@@ -7,6 +7,9 @@ class PagesController < ApplicationController
 
   def index
     @pages = Search::PageSearcher.new(params).search
+    @ascending = determine_ascending(params)
```

Comment:
Nice feature!

I think the additions to this controller are all concerns of the view (and not `PageController`). You could move the logic out into `PageHelper`.